java/test/eda/sorting/SortingTest.java

```diff
@@ -0,0 +1,86 @@
+package eda.sorting;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(value = Parameterized.class)
+public class SortingTest {
+
+    private Sorting sortingTest;
+
+    public SortingTest(Sorting sortingTest) {
+        this.sortingTest = sortingTest;
+    }
+
+    @Test
+    public void test_random(){
+        Integer[] h = new Integer[50000];
+        for(int i = 0; i<h.length;i++){
```

Comment:
You always create a random array and we can't see what is the expected array, aren't you? 
I think that if you only create a one method with random array with hight length is enough. 
Each other method should creates a -manual- array and We can see that the array {1,4,3} finally is sorted like this {1,3,4}.

But well played!!

![](http://gif.co/v1jR.gif)